### PR TITLE
mongodb_replicaset module and test exception

### PR DESCRIPTION
**Pull request by rhysmeister**
##### SUMMARY

New module allowing the creation of mongodb replicasets. Based on existing mongodb modules.

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

mongodb_replicaset

##### ANSIBLE VERSION

```
ansible 2.5.5
  config file = None
  configured module search path = [u'/Users/rhyscampbell/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.15 (default, Jun 17 2018, 12:46:58) [GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)]
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: lib/ansible/modules/database/mongodb/mongodb_replicaset.py [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+        required: false
+        default: 27017
+    replica_set:
+        description:
+            - Replicaset name
+        required: false
+        default: rs0
+    members:
+        description:
+            - An comma sepaeated list consisting of the replicaset members.
```
Comment (acozine) [inline lib/ansible/modules/database/mongodb/mongodb_replicaset.py]: This line should begin "A comma-separated list" ("a" not "an", add a hyphen to the compound adjective, typo in "separated").

File: lib/ansible/modules/database/mongodb/mongodb_replicaset.py [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+        default: True
+        type: bool
+    ssl:
+        description:
+            - Whether to use an SSL connection when connecting to the database
+        default: False
+        type: bool
+    ssl_cert_reqs:
+        description:
+            - Specifies whether a certificate is required from the other side of the connection, and whether it will be validated if provided.
```
Comment (acozine) [inline lib/ansible/modules/database/mongodb/mongodb_replicaset.py:70]: This line is a complete sentence and ends with a period. Some of the other `description` lines match this pattern; others don't. Make all `description` lines the same.

File: lib/ansible/modules/database/mongodb/mongodb_replicaset.py [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
+        required: false
+        default: "CERT_REQUIRED"
+        choices: ["CERT_REQUIRED", "CERT_OPTIONAL", "CERT_NONE"]
+    arbiter_at_index:
+        description: Identifies the position of the member in the array that is an arbiter.
+        required: false
+    chainingAllowed:
+        description: >
+            When settings.chainingAllowed is true, the replica set allows secondary members to replicate from other
```
Comment (acozine) [inline lib/ansible/modules/database/mongodb/mongodb_replicaset.py]: Elsewhere you make `replicaset` a single word. For consistency, please use one form or the other throughout.

File: lib/ansible/modules/database/mongodb/mongodb_replicaset.py [hunk 4]
```diff
@@ -0,0 +1,412 @@
...
+            secondary members. When settings.chainingAllowed is false, secondaries can replicate only from the primary.
```
Comment (acozine) [inline lib/ansible/modules/database/mongodb/mongodb_replicaset.py]: When referring to settings (for example, `settings.chainingAllowed` here and in the line above), you can make them stand out with formatting - see  https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html#linking-within-module-documentation for details.

File: lib/ansible/modules/database/mongodb/mongodb_replicaset.py [hunk 5]
```diff
@@ -0,0 +1,412 @@
...
+        default: true
+        required: false
+        type: bool
+    heartbeatTimeoutSecs:
+        description: >
+            Number of seconds that the replica set members wait for a successful heartbeat from each other.
```
Comment (acozine) [inline lib/ansible/modules/database/mongodb/mongodb_replicaset.py]: Here again, `replica set` vs `replicaset`.

File: lib/ansible/modules/database/mongodb/mongodb_replicaset.py [hunk 6]
```diff
@@ -0,0 +1,412 @@
...
+            If a member does not respond in time, other members mark the delinquent member as inaccessible.
+            The setting only applies when using protocolVersion 0. When using protocolVersion 1 the relevant
```
Comment (acozine) [inline lib/ansible/modules/database/mongodb/mongodb_replicaset.py]: Formatting for `protocolVersion`.

File: lib/ansible/modules/database/mongodb/mongodb_replicaset.py [hunk 7]
```diff
@@ -0,0 +1,412 @@
...
+            setting is settings.electionTimeoutMillis.
+        default: 10
+        required: false
+    electionTimeoutMillis:
+        description: The time limit in milliseconds for detecting when a replica set's primary is unreachable.
+        default: 10000
+        required: false
+    protocolVersion:
+        description: Version of the replica set election protocol.
```
Comment (acozine) [inline lib/ansible/modules/database/mongodb/mongodb_replicaset.py]: Another instance of `replica set` vs. `replicaset`.

File: lib/ansible/modules/database/mongodb/mongodb_replicaset.py [hunk 8]
```diff
@@ -0,0 +1,412 @@
...
+        default: 1
+        required: false
+        choices: [0,1]
+notes:
+    - Requires the pymongo Python package on the remote host, version 2.4.2+. This
+      can be installed using pip or the OS package manager. @see http://api.mongodb.org/python/current/installation.html
```
Comment (acozine) [inline lib/ansible/modules/database/mongodb/mongodb_replicaset.py]: That same docs page about formatting shows how to create a link for this URL: U(http://server.to.visit.com/address/page).

**Review by acozine — COMMENTED** [review]
I see a few places where the documentation for this module could be improved.

File: lib/ansible/modules/database/mongodb/mongodb_replicaset.py (hunk 8 above)
Comment (rhysmeister) [reply lib/ansible/modules/database/mongodb/mongodb_replicaset.py]: OK. Thanks for the feedback. I've committed a number of corrections.

Rhys

**Comment by gundalow** [comment]
@nerzhul @elliotttf @Lujeni I believe you've worked on other Mongo modules, your review would be welcome on this

File: lib/ansible/modules/database/mongodb/mongodb_replicaset.py [hunk 9]
```diff
@@ -0,0 +1,412 @@
...
+        default: 27017
+    replica_set:
+        description:
+            - Replicaset name.
+        required: false
+        default: rs0
+    members:
+        description:
+            - A comma-separated consisting of the replicaset members.
+            - Supply as a simple csv string, i.e. mongodb1:27017,mongodb2:27017,mongodb3:27017.
```
Comment (maxamillion) [inline lib/ansible/modules/database/mongodb/mongodb_replicaset.py:55]: Can this alternatively also be a YAML list? This is common in other modules (the package manager modules come to mind) and I'd like to see it carried over here if possible.

**Review by maxamillion — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/database/mongodb/mongodb_replicaset.py (hunk 9 above)
Comment (rhysmeister) [reply lib/ansible/modules/database/mongodb/mongodb_replicaset.py:55]: I've made some changes to allow the replicaset members to be supplied as a list and update the examples in the documentation section.

Got into a bit of a tiz with a rebase. Not sure why at the moment but it's cause the shippable tests to be cancelled.

**Comment by rhysmeister** [comment]
Will close this PR and resubmit as this one is a bit of a mess.

**Pull request closed without merging**